quic/s2n-quic-core/src/crypto/application/limited.rs

```diff
@@ -12,9 +12,30 @@ pub struct Key<K> {
     // Keeping encrypted_packets out of the key allow keys to be immutable, which allows optimizations
     // later on.
     encrypted_packets: u64,
+    decrypted_packets: u64,
     confidentiality_limit: u64,
 }
 
+#[derive(Copy, Clone, Debug)]
+#[non_exhaustive]
+pub struct Limits {
+    pub key_update_window: u64,
+    pub sealer_optimization_threshold: u64,
+    pub opener_optimization_threshold: u64,
+    pub max_mtu: MaxMtu,
+}
```

Comment:
Could you add some documentation for these?

Reply:
will do